addons/l10n_hr_edi/models/account_move_send.py

```diff
@@ -0,0 +1,126 @@
+        # EXTENDS 'account'
+        if 'mojeracun' in kwargs.get('sending_methods', []):
+            return 'ubl_hr'
+        return super()._get_default_invoice_edi_format(move, **kwargs)
+
+    def _is_applicable_to_company(self, method, company):
+        # EXTENDS 'account'
+        if method == 'mojeracun':
+            return company.l10n_hr_mer_proxy_state != 'rejected' and company.country_code == 'HR'
+        return super()._is_applicable_to_company(method, company)
+
+    def _is_applicable_to_move(self, method, move, **move_data):
+        # EXTENDS 'account'
+        if method == 'mojeracun':
+            partner = move.partner_id.commercial_partner_id.with_company(move.company_id)
+            invoice_edi_format = move_data.get('invoice_edi_format') or 'ubl_hr'
+            return all([
+                self._is_applicable_to_company(method, move.company_id),
+                partner.invoice_edi_format == 'ubl_hr',
+                partner.invoice_sending_method == 'mojeracun',
+                move.company_id.l10n_hr_mer_proxy_state != 'rejected',
+                move._need_ubl_cii_xml(invoice_edi_format)
+                or move.ubl_cii_xml_id and move.l10n_hr_mer_document_status not in {'20', '30', '40'},
+            ])
+        return super()._is_applicable_to_move(method, move, **move_data)
+
+    def _hook_if_errors(self, moves_data, allow_raising=True):
+        # EXTENDS 'account'
+        moves_failed_file_generation = self.env['account.move']
+        for move, move_data in moves_data.items():
+            if 'mojeracun' in move_data['sending_methods'] and move_data.get('blocking_error'):
+                moves_failed_file_generation |= move
+        moves_failed_file_generation.l10n_hr_mer_document_status = '45'
+        return super()._hook_if_errors(moves_data, allow_raising=allow_raising)
+
+    def _call_web_service_after_invoice_pdf_render(self, invoices_data):
+        print("--- DEBUG: O:ams - _call_web_service() ---")
+        # EXTENDS 'account'
+        super()._call_web_service_after_invoice_pdf_render(invoices_data)
+
+        for invoice, invoice_data in invoices_data.items():
+            print(" -> loop:", invoice, end=' ')
+            # Looks like MojEracun determines the receiver endpoint entirely from the XML
+            # (as there are no other parameters for the send API), so no need to check for partner endpoint
+            # The error retured is in Croatian, but a HR user is supposed to be able to read it
+            if 'mojeracun' not in invoice_data['sending_methods']:
+                print("- not in sending methods!")
+                continue
+            if not self._is_applicable_to_move('mojeracun', invoice, **invoice_data):
+                print("- not appliccable!")
+                continue
+            self._l10n_hr_send_to_mer(invoice, invoice_data)
+
+    def _l10n_hr_send_to_mer(self, invoice, invoice_data):
```

Comment:
Thanks for splitting this in a separate method! I would suggest, rather than passing the 'invoice_data' dict and assigning errors in it, to have the following signature:

```
def _l10n_hr_send_to_mer(self, invoice, xml_file):
    ...
    try:
        response = self.env['l10n_hr_edi.mojeracun_connection']._mer_api_send(invoice.company_id, xml_file.decode())
    except MojEracunServiceError as e:
        return e.message
    ...
```

That way the method returns the error if there is one.

I think it would make more sense to do it this way :smile: